plugins/references.py

```diff
@@ -760,6 +781,70 @@ def run(self, edit, name=None):
             offset -= convert2ref(view, edit, _link_span, link_span[1], link_span[2])
 
 
+class MdeConvertBareLinkToMdLinkCommand(MdeTextCommand):
+    """Convert an inline link to reference."""
+
+    def is_visible(self):
+        """Return True if selection contains links"""
+        view = self.view
+        for sel in view.find_by_selector(barelink_scope_name):
+            if any(s.intersects(sel) for s in view.sel()):
+                return True
+        return False
+
+    def run(self, edit, name=None):
+        """Run command callback."""
+        # import queue
+        import threading
+        import time
+
+        thread_queue = []
+
+        url_titles = {}
+        url_redirects = {}
+
+        def getTitleFromUrlJob(link_href):
+            import urllib.request
+
+            resp = urllib.request.urlopen(link_href)
```

Comment:
Some kind of cache control might be required to speed up things if several links point to the same page/target.